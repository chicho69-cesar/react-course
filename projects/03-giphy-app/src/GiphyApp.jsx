import { useState } from "react"
import { AddCategory } from "./components/AddCategory"
import { GiftGrid } from "./components/GiftGrid"

function GiphyApp() {
  const [categories, setCategories] = useState(['Dragon Ball'])

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return
    setCategories([newCategory, ...categories])
  }

  return (
    <>
      <h1>
        Giphy App!
      </h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GiftGrid
          key={category}
          category={category}
        />
      ))}
    </>
  )
}

export default GiphyApp
