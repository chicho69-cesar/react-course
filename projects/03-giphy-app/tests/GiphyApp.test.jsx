/* eslint-disable no-undef */
import { render, screen, fireEvent } from '@testing-library/react'
import GiphyApp from '../src/GiphyApp'

describe('Test on <GiphyApp />', () => {
  test('should render the component correctly', () => {
    const { container } = render(<GiphyApp />)
    expect(container).toMatchSnapshot()
  })

  test('should show the title of the app', () => {
    render(<GiphyApp />)
    expect(document.querySelector('h1').innerHTML).toBe('Giphy App!')
  })

  test('should render initial category', () => {
    render(<GiphyApp />)
    expect(screen.getByText('Dragon Ball'))
  })

  test('should show search category', () => {
    const inputValue = 'Goku'

    render(<GiphyApp />)

    const input = screen.getByRole('textbox')
    const form = screen.getByRole('form', { name: 'form' })

    fireEvent.input(input, { target: { value: inputValue } })
    fireEvent.submit(form)

    expect(input.value).toBe('')

    expect(screen.getByText(inputValue))
    expect(screen.getAllByText(inputValue).length).toBe(1)
  })
})
