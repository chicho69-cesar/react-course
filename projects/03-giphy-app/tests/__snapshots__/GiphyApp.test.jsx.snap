// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GiphyApp /> should render the component correctly 1`] = `
<div>
  <h1>
    Giphy App!
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifts"
      type="text"
      value=""
    />
  </form>
  <h3>
    Dragon Ball
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    aria-label="card-grid"
    class="card-grid"
  />
</div>
`;
